lib/logdna/client.rb

```diff
@@ -1,34 +1,53 @@
 # frozen_string_literal: true
 
+require "etc"
 require "net/http"
 require "socket"
 require "json"
 require "concurrent"
 require "date"
+require 'securerandom'
 
 module Logdna
+  Message = Struct.new(:source, :size)
+
   class Client
     def initialize(request, uri, opts)
       @uri = uri
 
       # NOTE: buffer is in memory
       @buffer = []
-      @buffer_byte_size = 0
-
-      @side_messages = []
 
       @lock = Mutex.new
-      @side_message_lock = Mutex.new
-      @flush_limit = opts[:flush_size] || Resources::FLUSH_BYTE_LIMIT
+      @flush_threshold = opts[:flush_size] || Resources::FLUSH_BYTE_LIMIT
+      @flush_limit = 16384
```

Comment:
where does this value come from, and should this be added to `Resources`?

Reply:
Decided to combine `@flush_limit` and `@flush_threshold` as having them separately was just confusing. This reuses the same option and `Resources::FLUSH_BYTE_LIMIT`.

I opted to change `FLUSH_BYTE_LIMIT` to 500KB 2 MiB to be consistent with the logic in the rust agent. That number works well there (probably?)

Reply:
I actually went a different route. I did the following renaming to be more clear:
1. `@flush_limit` => `@request_size`
2. `@flush_threshold` => `@flush_size`

Both default to 2 MiB make a differentiation between a flush and a request.